modules/extension/app-schema/app-schema/src/main/java/org/geotools/appschema/jdbc/JoiningJDBCFeatureSource.java

```diff
@@ -1458,22 +1448,33 @@ protected int getCountInternal(Query query) throws IOException {
         // the primaryKey from the wrapped JDBCSource
         String idColumnName = getIdColumnName(getSchema());
         JoiningQuery jQuery = (JoiningQuery) query;
-        FilterAttributeExtractor extractor = new FilterAttributeExtractor();
+
+        // Checks if the filter is based on root property
+        // and we can go with the root JdbcSource to count
+        // or we need a Join query.
+        // Uses a MultipleValueExtractor to detect
+        // if there are MultipleValues
+        MultipleValueExtractor extractor = new MultipleValueExtractor();
         jQuery.getRootMapping().getFeatureIdExpression().accept(extractor, null);
         String[] attributes = extractor.getAttributeNames();
         // the id from the app-schema configuration
         String idMapping = attributes.length > 0 ? extractor.getAttributeNames()[0] : null;
         boolean idsColumnEquals = idColumnName.equals(idMapping);
-        boolean isJoining = jQuery.getQueryJoins().size() > 0;
+        jQuery.getFilter().accept(extractor, null);
+        boolean isNestedFilter =
+                NestedFilterToSQL.isNestedFilter(jQuery.getFilter())
```

Comment:
This boolean building block is large enough to be a private method explaining how this work.

Reply:
fixed